addon/components/async-button.js

```diff
@@ -38,9 +38,18 @@ const ButtonComponent = Component.extend(positionalParamsMixin, {
       set(this, 'promise', promise);
     };
 
-    let actionArguments = ['action', callbackHandler, ...params];
+    if (typeof this.attrs.action === 'function') {
+      let promise = this.attrs.action();
```

Comment:
Opted for closure action currying rather than the `positionalParams` implementation. Should we still pass along `params` here?